### PR TITLE
feat: add customerOsId, referenceId to Org view

### PR DESCRIPTION
**Pull request by acalinica**
## Proposed changes
This PR adds customerOsId and referenceId along with a link to parent organization in the Organization Profile VIew. It also fixes some performance issue with the Table component and displays a new website column along with the parent organization name in the Company column.

## Changes

What types of changes does your code introduce?  _Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build related changes
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Other (please describe below)

### CODE REVIEW

File: packages/apps/spaces/app/organization/[id]/src/components/OrganizationSidenav/OrganizationSidenav.tsx [hunk 1]
```diff
@@ -57,29 +59,41 @@ export const OrganizationSidenav = () => {
...
-          <IconButton
-            size='xs'
-            variant='ghost'
-            aria-label='Go back'
-            onClick={() => {
-              router.push(`/${lastActivePosition?.root || 'organization'}`);
-            }}
-            icon={<Icons.ArrowNarrowLeft color='gray.700' boxSize='6' />}
-          />
+      <Flex gap='2' align='center' mb='4'>
```
Comment (Kasia9090) [inline packages/apps/spaces/app/organization/[id]/src/components/OrganizationSidenav/OrganizationSidenav.tsx:62]: Aria-label is for screen readers, it won't result in displaying tooltip which in this case is needed when text is truncated

File: packages/apps/spaces/app/organizations/page.tsx [hunk 2]
```diff
@@ -216,35 +206,21 @@ export default function OrganizationsPage() {
...
             table={table}
-            selection={selection}
-            onArchiveOrganizations={onOpen}
+            isArchiving={hideOrganizations.isLoading}
             onMergeOrganizations={handleMergeOrganizations}
+            onArchiveOrganizations={handleHideOrganizations}
           />
         )}
       />
-
```
Comment (Kasia9090) [inline packages/apps/spaces/app/organizations/page.tsx:236]: How this is handled now?

File: packages/apps/spaces/app/organizations/src/components/Actions/Actions.tsx [hunk 3]
```diff
@@ -1,58 +1,107 @@
...
-  );
-};
+          label={`Archive selected ${
+            selectCount === 1 ? 'organization' : 'organizations'
+          }?`}
+        />
+      </>
+    );
+  },
+  (prev, next) => {
```
Comment (Kasia9090) [inline packages/apps/spaces/app/organizations/src/components/Actions/Actions.tsx:129]: the condition is missing `isArchiving` check

File: packages/apps/spaces/app/layout.tsx [hunk 4]
```diff
@@ -21,7 +21,7 @@ const barlow = localFont({
       style: 'normal',
     },
   ],
-  preload: true,
+  preload: false,
```
Comment (Kasia9090) [inline packages/apps/spaces/app/layout.tsx:24]: why this change?

**Review by Kasia9090 — CHANGES_REQUESTED** [review]

File: packages/apps/spaces/app/layout.tsx (hunk 4 above)
Comment (acalinica) [reply packages/apps/spaces/app/layout.tsx:24]: This change is needed to make some warnings related about preloading go away from the console.

File: packages/apps/spaces/app/organizations/src/components/Actions/Actions.tsx (hunk 3 above)
Comment (acalinica) [reply packages/apps/spaces/app/organizations/src/components/Actions/Actions.tsx:129]: Actually I'll refactor this a bit to make it even better. I'll move mutation handlers inside the Actions compoment so that there's no need to pass these fns through props.

File: packages/apps/spaces/app/layout.tsx (hunk 4 above)
Comment (Kasia9090) [reply packages/apps/spaces/app/layout.tsx:24]: what were those warnings?

File: packages/apps/spaces/app/organizations/src/components/Actions/Actions.tsx (hunk 3 above)
Comment (Kasia9090) [reply packages/apps/spaces/app/organizations/src/components/Actions/Actions.tsx:129]: isArchiving is a boolean attr, not function, anyway let me know when it's done

**Review by Kasia9090 — APPROVED** [review]
